### ISSUE TITLE
covr thinks I am CRAN

### ISSUE THREAD

**Issue by wlandau** [r-lib/covr#314]
The `skip_on_cran()` tests are skipped even locally. This does not seem to happen with `devtools` or `testthat` alone. Reprex: https://github.com/wlandau/covrtest.

`R/f.R`:

```r
f <- function(){
  "x"
}
```

`tests/testthat/test-coverage.R`:

```r
context("test-coverage.R")

test_that("multiplication works", {
  skip_on_cran()
  expect_equal(f(), "x")
})
```

Testing:

```r
devtools::test("~/covrtest")
#> Loading covrtest
#> Testing covrtest
#> ✔ |  OK F W S | Context
#> 
⠏ |   0       | test-coverage.R
✔ |   1       | test-coverage.R
#> 
#> ══ Results ═══════════════════════════════════════════════════════════════════════════
#> OK:       1
#> Failed:   0
#> Warnings: 0
#> Skipped:  0
```

Coverage:

```r
covr::package_coverage("~/covrtest")
#> covrtest Coverage: 0.00%
#> R/f.R: 0.00%
```

Session:

```r
> sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux

Matrix products: default
BLAS: CENSORED/R-3.4.4/lib64/R/lib/libRblas.so
LAPACK: CENSORED/R-3.4.4/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] testthat_2.0.0.9000  usethis_1.3.0.9000   devtools_1.13.5.9000

loaded via a namespace (and not attached):
 [1] digest_0.6.15        crayon_1.3.4         withr_2.1.2
 [4] rprojroot_1.3-2      assertthat_0.2.0     R6_2.2.2
 [7] backports_1.1.2-9000 magrittr_1.5         rlang_0.2.0.9001
[10] debugme_1.1.0        cli_1.0.0            callr_2.0.3
[13] desc_1.2.0           compiler_3.4.4       pkgload_1.0.0
[16] pkgbuild_1.0.0       memoise_1.1.0
>
```

All on the latest commits of 

- `covr`: https://github.com/r-lib/covr/commit/b4e282e7f0befb37bb4e56fcecf181d4edfc0a77
- `devtools`: https://github.com/r-lib/devtools/commit/13ee56b9bb43dd2cb2e3b364bcba2c49ef350fac
- `testthat`: https://github.com/r-lib/testthat/commit/3064e6811dc8c2f2f9dc2f7142781092859f75b6

**Comment by jimhester** [comment]
covr is by design completely agnostic to testing framework. `NOT_CRAN` is a devtools / testthat convention, so you will need to set it manually if you want covr to respect it.

**Comment by wlandau** [comment]
Thanks, Jim! I added `Sys.setenv(NOT_CRAN = "true")` to my `.Rprofile`, and it solved the problem.

**Comment by jameslamb** [comment]
@jimhester thank you! I came here looking to report this same problem tonight. @wlandau appreciate you adding extra context on how to get testthat to play nice

**Comment by randy3k** [comment]
I also rediscovered this, and I used `withr::with_envvar` to fix it.

```r
withr::with_envvar(
	c("NOT_CRAN" = "true"), 
	covr::package_coverage())
`

**Comment by mccroweyclinton-EPA** [comment]
It makes sense that having the environment variable NOT_CRAN=FALSE would keep covr from reporting coverage properly, but I just spent a few weeks trying to figure out why covr stopped working for me? If it were not for the help of a colleague I would have probably never figured out why covr kept reporting 0.00% coverage, I must have set NOT_CRAN=FALSE to test something out sometime in the past and forgot that environment variable was set.

I am requesting that covr either throws a warning that reporting results will not be accurate if it detects that NOT_CRAN=FALSE or that this requirement be documented somewhere in covr's vignettes and README so that others don't experience the same frustration as I just did.
